connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/CustomSparkConnectBlockingStub.scala

```diff
@@ -18,13 +18,93 @@ package org.apache.spark.sql.connect.client
 
 import scala.jdk.CollectionConverters._
 
+import com.google.protobuf.GeneratedMessageV3
 import io.grpc.ManagedChannel
+import io.grpc.stub.StreamObserver
 
 import org.apache.spark.connect.proto._
 
+// This is common logic shared between the blocking and non-blocking stubs.
+//
+// The common logic is responsible to verify the integrity of the response. The invariant is
+// that the same stub instance is used for all requests from the same client.
+private[client] class SparkConnectCommonStub {
+  // Server side session ID, used to detect if the server side session changed. This is set upon
+  // receiving the first response from the server. This value is used only for executions that
+  // do not use server-side streaming.
+  private var serverSideSessionId: Option[String] = None
+
+  protected def verifyResponse[RespT <: GeneratedMessageV3](fn: => RespT): RespT = {
+    val response = fn
+    val field = response.getDescriptorForType.findFieldByName("server_side_session_id")
+    // If the field does not exist, we ignore this for now.
```

Comment:
Maybe log a log warning.

Better don't put "for now" in comment unless there's an actionable followup (when do we plan to do something about it?).

Reply:
Done